src/items/functions.md

```diff
@@ -56,7 +56,7 @@ r[items.fn.signature]
 Functions may declare a set of *input* [*variables*][variables] as parameters, through which the caller passes arguments into the function, and the *output* [*type*][type] of the value the function will return to its caller on completion.
 
 r[items.fn.implicit-return]
-If the output type is not explicitly stated, it is the [unit type].
+If the output type is not explicitly stated, it is the [unit type]. However, if the block expression is not [diverging](../divergence.md), then the output type is instead [`!`](../types/never.md).
```

Comment:
I don't understand this change. Can you explain what it is getting at?

We were initially thinking the `not` is wrong, but that also seems wrong since the implicit return type is still `()` for a function with a diverging block.

```rust
#![feature(never_type)]
pub fn f() {
    panic!("");
}

struct S {
    x: !
}

pub fn z() {
    S {x: f()}; // ERROR: expected `!`, found `()`
}
```